### PR TITLE
Added common implementation of triplet finding and weight updating

### PR DESCRIPTION
**Pull request by guilhermeAlmeida1**
Following #212 , a common implementation was now created for the 2 next steps in seed finding.

No significant difference in performance was observed for now, which was expected due to the poor performance of the get_prefix_sum function already observed in #217 #218 .

Sample performance output:
./bin/traccc_seq_example_sycl --detector_file=tml_detector/trackml-detector.csv --digitization_config_file=tml_detector/default-geometric-config-generic.json --input_directory=tml_full/ttbar_mu100/ --run_cpu=0 --events=10

**NEW:**
==> Elpased time ... 
wall time           34.8908   
file reading (cpu)        30.805    
clusterization_time (cpu) 0         
spacepoint_formation_time (cpu) 0         
clusterization_time (sycl) 0.566461  
seeding_time (cpu)        0         
seeding_time (sycl)       3.31363   
tr_par_esti_time (cpu)    0         
tr_par_esti_time (sycl)   0.205617

**OLD:**
wall time           34.6785   
file reading (cpu)        30.5384   
clusterization_time (cpu) 0         
spacepoint_formation_time (cpu) 0         
clusterization_time (sycl) 0.570876  
seeding_time (cpu)        0         
seeding_time (sycl)       3.36168   
tr_par_esti_time (cpu)    0         
tr_par_esti_time (sycl)   0.207476

### CODE REVIEW

File: device/common/include/traccc/seeding/device/weight_updating.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/// Function used for updating the triplets' weights
+///
+/// @param[in] globalIndex   The index of the current thread
+/// @param[in] filter_config        Seedfilter configuration
+/// @param[in] sp_view       The spacepoint grid
+/// @param[in] triplet_ps_view    Prefix sum for iterating over the triplets
+/// @param[out] triplet_view Container storing the triplets
+///
+TRACCC_HOST_DEVICE
+void update_weights(const std::size_t globalIndex,
```
Comment (krasznaa) [inline device/common/include/traccc/seeding/device/weight_updating.hpp:24]: The function name is okay like this. Though I would prefer to make it even a bit more explicit, and call it `traccc::device::update_triplet_weights`.

But importantly, make sure that the file names correspond to the function names. At first I thought the function would be called `traccc::device::weight_updating`...

File: device/common/include/traccc/seeding/device/weight_updating.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Project include(s).
```
Comment (krasznaa) [inline device/common/include/traccc/seeding/device/weight_updating.hpp:1]: Add the common copyright statement to all new files.

Also, add

```c++
#pragma once
```

to all of the header files. (Checking again, you only seem to have forgotten about both of them for this one file. :stuck_out_tongue:)

**Review by krasznaa — CHANGES_REQUESTED** [review]
Let's start with just these. I'll continue looking tomorrow. :wink:

**Comment by guilhermeAlmeida1** [comment]
Ignore this latest push for now, that was just rebasing off of the main branch. I will make the changes requested here tomorrow as well

**Review by krasznaa — CHANGES_REQUESTED** [review]
Most of them are **very** surface level requests. The only real change is with the `triplet` handling in the weight updating kernel. :wink:

File: device/common/include/traccc/seeding/device/impl/update_triplet_weights.ipp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                filter_config.deltaRMin) {
+                newCompSeed = false;
+                break;
+            }
+        }
+
+        if (newCompSeed) {
+            compat_seedR[num_compat_seedR] = otherTop_r;
+            triplets_per_bin[ps_idx.second].weight +=
+                filter_config.compatSeedWeight;
```
Comment (krasznaa) [inline device/common/include/traccc/seeding/device/impl/update_triplet_weights.ipp:114]: Ahh... This is not what I had in mind...

Let's use `this_triplet` in this line. Let's just update the weight in the thread-specific variable during the for-loop.

Then, at the very end of the function, after the for-loop, let's just write to the global memory once.

```c++
triplets_per_bin[ps_idx.second].weight = this_triplet.weight;
```

**Review by krasznaa — CHANGES_REQUESTED** [review]
Still one small correction...

**Review by krasznaa — APPROVED** [review]
Okay, as long as the CI succeeds, and you say that the code works correctly like this, let's go with this version...

**Pull request merged**